src/devices/Camera/Camera/ProcessRunner.cs

```diff
@@ -0,0 +1,220 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Pipelines;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Iot.Device.Camera.Settings;
+
+namespace Iot.Device.Camera;
+
+/// <summary>
+/// Controls an external process
+/// </summary>
+public class ProcessRunner : IDisposable
+{
+    private readonly ProcessSettings _processSettings;
+    private CancellationTokenSource _cts;
+    private Process? _process = null;
+
+    /// <summary>
+    /// Creates an instance of the ProcessRunner by getting the
+    /// Process Settings
+    /// </summary>
+    public ProcessRunner(ProcessSettings processSettings)
+    {
+        _cts = new CancellationTokenSource();
+        _processSettings = processSettings;
+    }
+
+    /// <summary>
+    /// The PID of the process that is running or null
+    /// </summary>
+    public int? PId => _process?.Id;
+
+    /// <summary>
+    /// Dispose the active running process
+    /// </summary>
+    public void Dispose()
+    {
+        _cts.Cancel();
+        Kill();
+    }
+
+    /// <summary>
+    /// Kill the process
+    /// </summary>
+    public void Kill()
+    {
+        try
+        {
+            if (_process != null && !_process.HasExited)
+            {
+#if NETSTANDARD2_0
+                _process.Kill();
+#else
+                _process.Kill(true);
+#endif
+                if (!_process.HasExited)
+                {
+                    _process.WaitForExit(5000);
+                }
+
+                _process.Dispose();
+            }
+        }
+        catch (Exception)
+        {
```

Comment:
The exception being catched here is because the process may exit by itself but at tht same time the user try to kill it.
Unfortunately logging here does not help much because the process can exit for many different reasons and they are not reported in the exception.